bencher/plotting/plot_filter.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+from bencher.bench_cfg import PltCntCfg, BenchCfg
+from bencher.bench_vars import ParametrizedSweep
+from dataclasses import dataclass
+
+
+class VarRange:
+    def __init__(self, lower_bound: int = 0, upper_bound: int = -1) -> None:
+        """A VarRange represents the bounded and unbounded ranges of integers.  This class is used to define filters for various variable types.  For example by defining cat_var = VarRange(0,0), calling matches(0) will return true, but any other integer will not match.  You can also have unbounded ranges for example VarRange(2,None) will match to 2,3,4... up to infinity. for By default the lower and upper bounds are set to -1 so so that no matter what value is passsed to matches() will return false.  Matches only takes 0 and positive integers.
+
+        Args:
+            lower_bound (int, optional): The smallest acceptable value to matches(). Passing None will result in a lower bound of 0 (as matches only accepts positive integers). Defaults to 0.
+            upper_bound (int, optional): The largest acceptable value to matches().  Passing None will result in no upper bound. Defaults to -1.
+        """
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+    def matches(self, val: int) -> bool:
+        """Checks that a value is within the variable range.  lower_bound and upper_bound are inclusive (lower_bound<=val<=upper_bound )
+
+        Args:
+            val (int): A positive integer representing a number of items
+
+        Returns:
+            bool: True if the items is within the range, False otherwise.
+        """
+        if val >= 0:
```

Comment:
This might look clearer if we have         

`if val < 0:
raise ValueError("val must be >= 0")

...
return lower_match and upper_match
`

Reply:
if val >= 0 is no longer needed since the complement is caught in the previous if